service/storage_service.cc

```diff
@@ -4706,12 +4706,11 @@ storage_service::describe_ring_for_table(const sstring& keyspace_name, const sst
         }
         for (auto& r : replicas) {
             dht::endpoint_details details;
-            auto& hostid = r.host;
-            auto endpoint = host2ip(hostid);
-            details._datacenter = topology.get_datacenter(hostid);
-            details._rack = topology.get_rack(hostid);
-            details._host = endpoint;
-            tr._rpc_endpoints.push_back(_gossiper.get_rpc_address(endpoint));
+            const auto& node = topology.get_node(r.host);
```

Comment:
I didn't understand the commit message. Is this patch fixing a pre-existing bug (not related to the main PR)? What happened before your fix, reads from system.token_ring would sometimes fail depending on which shard did it?

Reply:
Yes, calls to this method would crash scylla if called from a non-0 shard.